### PR TITLE
Update authentication and user profile functionality

### PR DESCRIPTION
**Pull request by wekesa360**
This pull request addresses issue #37 and issue #36 

This pull request includes updates to the authentication and user profile functionality. It updates the `requirements.txt` file to include `djangorestframework-simplejwt 5.3.1`, updates the `DJANGO_SETTINGS_MODULE` in `asgi.py` and `wsgi.py`, updates the URLs in `urls.py` and `accounts/urls.py`, and updates `base.py` with REST framework and JWT configurations. It also refactors the `token_generator_and_check_if_exists` function in `utils.py` and handles exceptions in `AuthUserSerializer` in `serializers.py`. Additionally, it fixes formatting issues in `accounts/views.py`.

### CODE REVIEW

File: back-end/accounts/views.py [hunk 1]
```diff
@@ -1,48 +1,115 @@
...
     )
+    @action(methods=["POST"], detail=False)
     def login(self, request):
         serializer = self.get_serializer(data=request.data)
-        serializer.is_valid(raise_exception=True)
-        user = get_and_authenticate_user(**serializer.validated_data)
-        data = AuthUserSerializer(user).data
-        return Response(data=data, status=status.HTTP_200_OK)
+        try:
+            serializer.is_valid(raise_exception=True)
```
Comment (nafdev01) [inline back-end/accounts/views.py:82]: add logic to validate the token before login

Comment (wekesa360) [reply back-end/accounts/views.py:82]: to access the login view  a user is not supposed to be authenticated, thus there is not token to be validated... it uses [AllowAny] permission class

File: back-end/accounts/views.py [hunk 2]
```diff
@@ -51,61 +118,407 @@ def login(self, request):
...
+            "200": SuccessSerializer,
+            "400": ErrorSerializer,
+            "500": ErrorSerializer,
+        },
+        description="Verify user email",
+    )
+    @action(methods=["GET"], detail=False)
+    def verify_email(self, request):
+        token = request.query_params.get("token")
+        serializer = self.get_serializer_class(data={"token": token})
```
[the remark this answers is not in the document]
Comment (wekesa360) [reply back-end/accounts/views.py:173]: It still works, I had earlier defined the `get_serializer_class()` prior to the refinement, as the last class method, reviewing and refactoring that.

[the remark this answers is not in the document]
Comment (nafdev01) [reply back-end/accounts/views.py:173]: the `get_serializer_class`  does not accept any arguments apart from self, so it results in a type error when you try to pass the `data` keyword arguement. It also doesn't handle initialization which is what the highlighted line of code is trying to do.

**Comment by nafdev01** [comment]
add the media directory to .gitignore

**Review by nafdev01 — APPROVED** [review]
looks good

**Pull request merged**